README.md

```diff
@@ -247,3 +283,101 @@ The workflow for continuous integration:
 * CI deploys changes to the staging environment (stage.app.io) in the AWS prod account.
 * Changes are promoted (or merged) to the git prod branch.
 * CI deploys changes to the prod environment (prod.app.io) in the AWS prod account.
+
+# Creation/Forwarding of OpenTelemetry data
+Schematic has been instrumented with a mix of
+[automationally instrumented libraries](https://github.com/open-telemetry/opentelemetry-python-contrib/tree/main/instrumentation)
+and [manual traces](https://opentelemetry-python.readthedocs.io/en/latest/api/trace.html).
+In addition it's been configured at startup to [conditionally turn on trace/log exports](https://github.com/Sage-Bionetworks/schematic/blob/778bf54db9c5b4de0af334c4efe034b3dde0b348/schematic/__init__.py#L82-L139)
+depending on how a few environment variables are set. The combination of these items let
+the schematic container running in ECS export telemetry data out of the container to be
+ingested somewhere else for long-term storage.
+
+
+Schematic is configured to send it's telemetry data to the OpenTelemetry Collector
+which then handles forwarding that data on to it's final destination. This is
+accomplished by setting a few environment variables on the Schematic container such as:
+
+```python
+from src.service_props import ServiceProps
+
+telemetry_environment_variables = {
+    "TRACING_EXPORT_FORMAT": "otlp",
+    "LOGGING_EXPORT_FORMAT": "otlp",
+    "TRACING_SERVICE_NAME": "schematic",
+    "LOGGING_SERVICE_NAME": "schematic",
+    "DEPLOYMENT_ENVIRONMENT": environment,
+    "OTEL_EXPORTER_OTLP_ENDPOINT": "http://otel-collector:4318",
+}
+
+app_service_props = ServiceProps(
+    container_name="schematic-app",
+    container_location="ghcr.io/sage-bionetworks/app:v1.0"
+    container_port=443,
+    container_memory=1024,
+    container_env_vars=telemetry_environment_variables,
+)
+```
+
+
+## OpenTelemetry Collector
+The OpenTelemetry collector is deployed into ECS and is running in
+[Gateway mode](https://opentelemetry.io/docs/collector/deployment/gateway/). This
+configuration allows for a single collector to be the central point for all telemetry
+data leaving the context of this deployed infrastructure. This central point allows us
+to configure where [authorization can be attached](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/extension/oauth2clientauthextension),
+[requests batched up](https://github.com/open-telemetry/opentelemetry-collector/blob/main/processor/batchprocessor/README.md), or [sensitive data be stripped](https://docs.honeycomb.io/send-data/opentelemetry/collector/handle-sensitive-information/).
+
+
+The configuration of all of these elements stems from [supplying a `config.yaml` file](https://opentelemetry.io/docs/collector/configuration/) as
+an environment variable to the otel collector container at startup. This config file is
+set up to be sourced from AWS Secrets manager. To accomplish this a filled out copy of
```

Comment:
there are no secrets in this configuration file, why is it using the secrets manager?

Reply:
ahh ok.